### PR TITLE
#641 - Repair and import tools are inaccessible when the platform is network activated

### LINKED ISSUE #641: Repair and import tools are inaccessible when the platform is network activated
**Describe the bug**
Repair and import tools are inaccessible when the platform is network activated

**To Reproduce**
1. In Multisite environment, network activate BuddyBoss Platform
2. Go to Network admin dashboard > BuddyBoss > Tools > Repair Community, you will notice that it is not properly displaying.
3. Go to Network admin dashboard > BuddyBoss > Tools, and try to navigate on the following tools:
- Repair Forums 
- Import Profile Types 
- Import Forums
- Import Media
You will receive an error message "Sorry, you are not allowed to access this page."

**Expected behavior**
Tools should be available even when network activated.

**Screenshots**
- Network admin dashboard > BuddyBoss > Tools > Repair Community
tools2

- Network admin dashboard > BuddyBoss > Tools
-- Repair Forums 
-- Import Profile Types 
-- Import Forums
-- Import Media
tools1

**Support ticket links**
https://secure.helpscout.net/conversation/1111909529/405/

### PR DESCRIPTION
**Pull request by janakkaneriya007**
Repair and import tools are inaccessible when the platform is network activated issue fixed

Fixed the Repair Community menu for multisite

### All Submissions:

* [x] Have you followed the [Pull Requests Guidelines](https://github.com/buddyboss/buddyboss-platform/wiki/Submitting-Pull-Requests#pull-request-guidelines)?
* [x] Does your code follow the [WordPress' Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Fixes #641 .

### How to test the changes in this Pull Request:

1. In Multisite environment, network activate BuddyBoss Platform
2. Go to Network admin dashboard > BuddyBoss > Tools > Repair Community, you will notice that it is not properly displaying.
3. Go to Network admin dashboard > BuddyBoss > Tools, and try to navigate on the following tools:

- Repair Forums
- Import Profile Types
- import Forums
- Import Media

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: src/bp-forums/admin/tools.php [hunk 1]
```diff
@@ -179,6 +222,24 @@ function bbp_admin_repair_list() {
...
 }
 
+/**
+ * Function get user's sub sites.
+ *
+ * @param int $user_id user id to get the sub sites
+ *
+ * @return mixed|void
+ */
+function bbp_get_blogs_list( $user_id = 0 ) {
```
Comment (mehulkaklotar) [inline src/bp-forums/admin/tools.php]: @janakkaneriya007 use https://developer.wordpress.org/reference/functions/get_sites/ here

File: src/bp-core/admin/js/settings-page.js [hunk 2]
```diff
@@ -692,6 +699,7 @@
 									'action': 'bp_admin_forum_repair_tools_wrapper_function',
 									'type': BbToolsForumsRepairActions[currentAction],
 									'offset': offset,
+									'site_id': SiteId,
```
Comment (mehulkaklotar) [inline src/bp-core/admin/js/settings-page.js]: @janakkaneriya007 `site_id` is never used in the backend in this PR. is it working?

Comment (mehulkaklotar) [reply src/bp-core/admin/js/settings-page.js]: why do we need global variable for site id?  just get the value of site from the dropdown `$('#select-site').val()` when clicked button or doing ajax request.

File: src/bp-forums/admin/tools.php (hunk 1 above)
Comment (mehulkaklotar) [reply src/bp-forums/admin/tools.php]: @janakkaneriya007 what is the use of `$user_id`? please remove not using codes.

Comment (mehulkaklotar) [reply src/bp-forums/admin/tools.php]: change the name of the function to bbp_get_sites()

File: src/bp-forums/admin/tools.php [hunk 3]
```diff
@@ -179,6 +222,17 @@ function bbp_admin_repair_list() {
 	return (array) apply_filters( 'bbp_repair_list', $repair_list );
 }
 
+/**
+ * Function get network sites.
+ *
```
Comment (mehulkaklotar) [inline src/bp-forums/admin/tools.php]: @janakkaneriya007 add `@since BuddyBoss 1.2.10`

File: src/bp-core/admin/bp-core-admin-functions.php [hunk 4]
```diff
@@ -2891,6 +2884,29 @@ function bp_import_profile_types_admin_menu() {
 		'bp_member_type_import_submenu_page'
 	);
 
+	if ( current_user_can( 'bbp_tools_page' ) ) {
+		if ( current_user_can( 'bbp_tools_repair_page' ) ) {
```
Comment (mehulkaklotar) [inline src/bp-core/admin/bp-core-admin-functions.php:2888]: @janakkaneriya007 
```suggestion
	if ( current_user_can( 'bbp_tools_page' ) && current_user_can( 'bbp_tools_repair_page' ) ) {
```

File: src/bp-core/admin/bp-core-admin-functions.php [hunk 5]
```diff
@@ -2891,6 +2884,29 @@ function bp_import_profile_types_admin_menu() {
...
+			add_submenu_page(
+				'buddyboss-platform',
+				__( 'Import Forums', 'buddyboss' ),
+				__( 'Forum Import', 'buddyboss' ),
+				'manage_options',
+				'bbp-converter',
+				'bbp_converter_settings'
+			);
+		}
+	}
```
Comment (mehulkaklotar) [inline src/bp-core/admin/bp-core-admin-functions.php:2907]:
```suggestion
```

**Comment by ankitguptaindia** [comment]
> @janakkaneriya007 👍
> 
> @ankitguptaindia can you test this one?

Tested. Working Fine.

https://www.loom.com/share/be84ba24e98a4204ab09d7bdd96c2f0a

**Pull request merged**